gym_microrts/petting_zoo_api.py

```diff
@@ -0,0 +1,162 @@
+        self.rewards = {agent: 0 for agent in self.agents}
+        self._cumulative_rewards = {agent: 0 for agent in self.agents}
+        self.dones = {agent: False for agent in self.agents}
+        self.infos = {agent: {} for agent in self.agents}
+        self.state = {agent: None for agent in self.agents}
+        self.observations = {agent: None for agent in self.agents}
+        self.num_moves = 0
+
+        self._agent_selector = agent_selector(self.agents)
+        self.agent_selection = self._agent_selector.next()
+
+    def step(self, action):
+        if self.dones[self.agent_selection]:
+            return self._was_done_step(action)
+
+        agent = self.agent_selection
+
+        # the agent which stepped last had its _cumulative_rewards accounted for
+        # (because it was returned by last()), so the _cumulative_rewards for this
+        # agent should start again at 0
+        self._cumulative_rewards[agent] = 0
+
+        # stores action of current agent
+        self.state[self.agent_selection] = action
+
+        if self._agent_selector.is_last():
+            # Step environment
+            actions_list = list(self.state.values())
+            actions = np.stack(actions_list, axis=0)
+            self.step_async(actions)
+            obs, reward, done, info = self.step_wait()
+
+            for i, agent in enumerate(self.agents):
+                self.rewards[agent] = reward[i]
+                self.dones[agent] = bool(done[i].astype)
+                self.observations[agent] = obs[i, :]
+
+            self.num_moves += 1
+        else:
+            self._clear_rewards()
+
+        # selects the next agent.
+        self.agent_selection = self._agent_selector.next()
+        # Adds .rewards to ._cumulative_rewards
+        self._accumulate_rewards()
+
+    def observe(self, agent):
+        '''
+        Returns the observation an agent currently can make. `last()` calls this function.
+        '''
+        agent_id = self.agent_name_mapping[agent]
+        obs = self.obs[agent_id, :, :, :]
+
+        return obs
+
+
+def main():
+    opponents = [microrts_ai.coacAI for _ in range(1)]
+
+    env = PettingZooMicroRTSGridModeSharedMemVecEnv(2, 1, ai2s=opponents)
+    # set_trace()
+    from pettingzoo.test import api_test
+    api_test(env, num_cycles=10, verbose_progress=True)
+
+    # actions = np.array([env.agent_action_space.sample(),
+    #                     env.agent_action_space.sample(), env.agent_action_space.sample()])
+    # actions = actions.reshape(3, env.width * env.height, env.action_dim)
+
+    # for agent in env.agent_iter():
+    #     observation, reward, done, info = env.last()
+    #     print(done)
+    #     if done:
+    #         break
+
+    #     agent_id = env.agent_name_mapping[agent]
+    #     action = actions[agent_id, :]
+    #     env.step(action)
+
+    #     if env.num_moves > 10:
+    #         break
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Please move this part of the code to the root directory in a file named `hello_world_pettingzoo.py`